proxy/src/console/mgmt.rs

```diff
@@ -88,20 +99,22 @@ async fn try_process_query(pgb: &mut PostgresBackendTCP, query: &str) -> Result<
     let resp: KickSession = serde_json::from_str(query).context("Failed to parse query as json")?;
 
     let span = info_span!("event", session_id = resp.session_id);
-    let _enter = span.enter();
-    info!("got response: {:?}", resp.result);
-
-    match notify(resp.session_id, Ok(resp.result)) {
-        Ok(()) => {
-            pgb.write_message_noflush(&SINGLE_COL_ROWDESC)?
-                .write_message_noflush(&BeMessage::DataRow(&[Some(b"ok")]))?
-                .write_message_noflush(&BeMessage::CommandComplete(b"SELECT 1"))?;
-        }
-        Err(e) => {
-            error!("failed to deliver response to per-client task");
-            pgb.write_message_noflush(&BeMessage::ErrorResponse(&e.to_string(), None))?;
+    async move {
+        info!("got response: {:?}", resp.result);
+
+        match notify(resp.session_id, Ok(resp.result)) {
+            Ok(()) => {
+                pgb.write_message_noflush(&SINGLE_COL_ROWDESC)?
+                    .write_message_noflush(&BeMessage::DataRow(&[Some(b"ok")]))?
+                    .write_message_noflush(&BeMessage::CommandComplete(b"SELECT 1"))?;
+            }
+            Err(e) => {
+                error!("failed to deliver response to per-client task");
+                pgb.write_message_noflush(&BeMessage::ErrorResponse(&e.to_string(), None))?;
+            }
         }
+        Ok(())
     }
```

Comment:
If this is a consistency thing "no use of span.enter()", I can accept this, but there's no use of await in this async block.

Maybe we can use [`Span::in_scope`](https://docs.rs/tracing/0.1.38/tracing/struct.Span.html#method.in_scope) instead?

Reply:
Yeah I was surprised tbh there is no flush here but got hung up on the `defer` on the task creation. Why don't we have to flush the socket? Probably because the caller handles that? I'll just restore this, make it non-sync, as it is quite confusing that we have an async without any awaits.

Reply:
Made non-async and reverted for this function in [f37898e](https://github.com/neondatabase/neon/pull/4612/commits/f37898e377b0d576f194c0526bb89183a765ae48)